### PR TITLE
Only upload files for rw filesystems

### PR DESCRIPTION
**Pull request by lukemarsden**
Only upload files for read-write filesystems. Fixes breakage of unmounting output repos.

### CODE REVIEW

File: src/server/pfs/cmds/mount_unix.go [hunk 1]
```diff
@@ -129,10 +130,24 @@ func mountCmds() []*cobra.Command {
 			if err != nil {
 				return err
 			}
+			defer c.Close()
+
+			user, err := c.WhoAmI(c.Ctx(), &auth.WhoAmIRequest{})
+			if err != nil {
+				if strings.Contains(fmt.Sprintf("%s", err), "the auth service is not activated") {
```
Comment (albscui) [inline src/server/pfs/cmds/mount_unix.go:137]: Maybe try using `errors.Is`
```suggestion
				if errors.Is(err, auth.ErrNotActivated) {
```

File: src/server/pfs/cmds/mount_unix.go [hunk 2]
```diff
@@ -129,10 +130,24 @@ func mountCmds() []*cobra.Command {
...
+					logrus.Info("Connected to Pachyderm cluster where auth is disabled")
```
Comment (albscui) [inline src/server/pfs/cmds/mount_unix.go:138]:
```suggestion
					logrus.Infof("Connected to Pachyderm cluster where auth is disabled")
```

File: src/server/pfs/cmds/mount_unix.go [hunk 3]
```diff
@@ -129,10 +130,24 @@ func mountCmds() []*cobra.Command {
...
+				} else {
+					// exit process if we can't talk to pachyderm, whoever
+					// instantiates us is responsible for retrying
+					return fmt.Errorf("error calling WhoAmI to test Pachyderm connection: %s", err)
```
Comment (albscui) [inline src/server/pfs/cmds/mount_unix.go:142]:
```suggestion
					return errors.EnsureStack(err)
```

**Review by albscui — COMMENTED** [review]
Just need to update some errror handling and logging code.

**Review by albscui — APPROVED** [review]
LGTM, the `ErrNotActivated` check is nit, but I think we should start using `errors.Is` more often.

**Review by smalyala — APPROVED** [review]

**Comment by smalyala** [comment]
So we have to pick and choose eh? Either successful mounting of repos with multiple branches or fixing the `ls` and `cp` bugs.

**Comment by lukemarsden** [comment]
For now!

**Pull request merged**